kv_store/lww_kvs/kvs_memory_server.cpp

```diff
@@ -437,28 +459,34 @@ int main(int argc, char* argv[]) {
         int ger = req.tuple(i).global_ebs_replication();
         cout << "Received a key request for key " + key + ".\n";
 
-        communication::Key_Response_Tuple* tp = res.add_tuple();
-        tp->set_key(key);
-
         // fill in placement metadata only if not already exist
         if (placement.find(key) == placement.end()) {
           placement[key] = key_info(gmr, ger);
         }
-        
-        // for now, randomly select a memory thread
-        int tid = 1 + rand()%MEMORY_THREAD_NUM;
-        communication::Key_Response_Address* tp_addr = tp->add_address();
-        if (sender == "server") {
-          tp_addr->set_addr(worker_node_t(ip, tid).id_);
-        } else {
-          tp_addr->set_addr(worker_node_t(ip, tid).proxy_connection_connect_addr_);
+
+        // check if the node is responsible for this key
+        auto result = responsible<master_node_t, crc32_hasher>(key, placement[key].global_memory_replication_, global_memory_hash_ring, mnode.id_);
+
+        if (result.first) {
```

Comment:
This is to make sure we didn't delete then receive gossip, right? We should comment that in.